src/scalper.ts

```diff
@@ -182,64 +257,41 @@ export class Scalper {
         {
           orderType: "limit",
           expiryTimestamp: getUnixTs() + twapInterval - 1,
-          clientOrderId: orderId,
+          clientOrderId: deltaOrderId,
         }
       );
+      hedgeCount++;
+      // Wait the twapInterval of time to see if WS gets any fill message
       console.log(
         this.symbol,
-        hedgeSide,
-        "#",
-        hedgeCount,
-        "-",
-        orderId,
-        "Size:",
-        Math.abs(hedgeDeltaClip),
-        "Price:",
-        hedgePrice
-      );
-      // Reduce hedge by what actually got filled
-      let filledSize = await fillSize(perpMarket, this.connection, orderId);
-      hedgeDeltaTotal = hedgeDeltaTotal + filledSize;
-      console.log(
-        this.symbol,
-        "Filled",
-        hedgeSide,
-        "Size",
-        filledSize,
-        "Remaining Size ",
-        hedgeDeltaTotal
-      );
-
-      // No need to wait for the twap interval if filled
-      if (Math.abs(hedgeDeltaTotal * fairValue) < (this.tickSize * fairValue)) {
-        break;
-      }
-      // Wait the twapInterval of time before sending updated hedge price & qty
-      console.log(
-        this.symbol,
-        "Delta Hedge",
-        hedgeCount + 1,
-        "Wait:",
+        "Scan Delta Fills for",
         twapInterval,
         "seconds"
       );
-      await sleepTime(twapInterval);
-
-      // Update Price
-      [mangoCache] = await loadPrices(mangoGroup, this.connection);
-      fairValue = mangoGroup.getPrice(this.marketIndex, mangoCache).toNumber();
-
-      // Keep count of # of hedges & create new orderID
-      orderId++;
-      hedgeCount++;
-    }
-    console.log(this.symbol, "Delta Hedge Complete");
+      for (let i=0; i<twapInterval; i++){
+        // Check every second if further Delta Hedging required
+        if (Math.abs(hedgeDeltaTotal * fairValue) < (this.tickSize * fairValue)) {
+          fillFeed.removeEventListener('message', deltaFillListener);
+          console.log(this.symbol, "Delta Hedge Complete");
+          return;
+        }
+        await sleepTime(1);
```

Comment:
Why do we still need a sleep if we are using listeners?

Reply:
We are returning the hedgeDetalTotal (remaining delta) once we get a fill, but we will not terminate the delta hedge until we bring this remaining required hedge to 0. If I embed this in the message listening it was breaking because we want a twap to occur. We may never get a delta hedge fill if the market is wide or illiquid, and will just get stuck here and never force the hedge through to then get to gamma scalp. 

There could be a better way here, but effectively were balancing the need to TWAP on a set timer with receiving fills async

Reply:
Again this is solving mostly for an edge case where the size is too big or market super illiquid. Expectation is 99% of the time we hedge the full amount in a single clip with a taker limit order

Reply:
Alright. Then make the 1 into a descriptive constant so it is clear that we need to wait in case we dont get a fill